c10/core/TensorImpl.h

```diff
@@ -2439,6 +2439,11 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     return has_symbolic_sizes_strides_;
   }
 
+  bool may_have_symbolic_sizes_strides() const {
```

Comment:
> So in this change, we add may_have_symbolic_sizes_strides() - which should never return false if the tensor has symbolic sizes and strides

If this is strictly never, should it be named more strictly?

Reply:
Hmm, in my opinion this naming more sense, e.g.:

`tensor.definitely_has_symbolic_sizes_strides()` -> if it returns true, we know that this tensor definitely has symbolic sizes/strides

`tensor.may_have_symbolic_sizes_strides()` -> if this returns true, then maybe the tensor has symbolic sizes/strides; but if this returns false, then we knwo this tensor definitely doesn't have symbolic sizes/strides.

I guess we could call it `tensor.definitely_does_not_have_symbolic_sizes_strides()` but that's a bit long...

Reply:
I would agree that if we're saying we have 3 states now: definitely yes, maybe, definitely no, it would make sense for the existing `has_symbolic_sizes_strides`  to be renamed to reflect what it is doing.

Reply:
ok, named it "does_not_have_symbolic_sizes_strides" with a comment explaining (and inverted the return value)